lib/eventsource.js

```diff
@@ -81,7 +81,10 @@ function EventSource (url, eventSourceInitDict) {
   var reconnectUrl = null
 
   function connect () {
-    var options = parse(url)
+    var options = url
+    if (typeof url === 'string') {
```

Comment:
This change makes sense; the spec does say that a [URL record](https://html.spec.whatwg.org/multipage/server-sent-events.html#eventsourceinit) is acceptable. Would you mind added a test for this?